src/vivarium/testing_utilities.py

```diff
@@ -148,31 +149,62 @@ def _non_crn_build_population(
     return population
 
 
-def build_table(value, year_start, year_end, columns=("age", "year", "sex", "value")):
-    value_columns = columns[3:]
+def build_table(
+    value,
+    parameter_columns={
+        "age": (0, 125),
+        "year": (1990, 2020),
+    },
+    key_columns={"sex": ("Female", "Male")},
+    value_columns=["value"],
+):
     if not isinstance(value, list):
         value = [value] * len(value_columns)
 
     if len(value) != len(value_columns):
         raise ValueError("Number of values must match number of value columns")
 
+    # Get product of parameter columns
+    range_parameter_product = {
+        key: list(range(value[0], value[1])) for key, value in parameter_columns.items()
+    }
+    # Build out dict of items we will need cartesian product of to make dataframe
+    product_dict = dict(range_parameter_product)
+    product_dict.update(key_columns)
+    products = product(*product_dict.values())
+
     rows = []
-    for age in range(0, 140):
-        for year in range(year_start, year_end + 1):
-            for sex in ["Male", "Female"]:
-                r_values = []
-                for v in value:
-                    if v is None:
-                        r_values.append(np.random.random())
-                    elif callable(v):
-                        r_values.append(v(age, sex, year))
-                    else:
-                        r_values.append(v)
-                rows.append([age, age + 1, year, year + 1, sex] + r_values)
+    for p in products:
```

Comment:
nit: Maybe it's a me thing, but I really dislike single-letter variable names. At least they're not `s`, `n`, or `c` (which mucks up pdb)